sktime/forecasting/compose/_reduce.py

```diff
@@ -2,108 +2,176 @@
 
-__author__ = "Markus Löning"
+__author__ = ["Lovkush Agarwal", "Markus Löning"]
 __all__ = [
-    "ReducedTabularRegressorMixin",
-    "ReducedTimeSeriesRegressorMixin",
+    "make_reduction",
     "DirectTimeSeriesRegressionForecaster",
     "RecursiveTimeSeriesRegressionForecaster",
-    "DirectRegressionForecaster",
-    "MultioutputRegressionForecaster",
-    "RecursiveRegressionForecaster",
+    "MultioutputTimeSeriesRegressionForecaster",
+    "DirectTabularRegressionForecaster",
+    "RecursiveTabularRegressionForecaster",
+    "MultioutputTabularRegressionForecaster",
     "ReducedForecaster",
+    "ReducedRegressionForecaster",
 ]
 
 import numpy as np
-import pandas as pd
+from sklearn.base import RegressorMixin
 from sklearn.base import clone
 
+from sktime.forecasting.base import ForecastingHorizon
 from sktime.forecasting.base._base import DEFAULT_ALPHA
 from sktime.forecasting.base._sktime import _BaseWindowForecaster
 from sktime.forecasting.base._sktime import _OptionalForecastingHorizonMixin
 from sktime.forecasting.base._sktime import _RequiredForecastingHorizonMixin
-from sktime.forecasting.model_selection import SlidingWindowSplitter
+from sktime.regression.base import BaseRegressor
+from sktime.utils._maint import deprecated
 from sktime.utils.validation import check_window_length
 from sktime.utils.validation.forecasting import check_step_length
-from sktime.utils.validation.forecasting import check_y
 
 
-##############################################################################
-# base classes for reduction from forecasting to regression
+def _concat_y_X(y, X):
+    """Helper function to concatenate y and X in preparation for sliding-window
+    transform"""
+    z = y.to_numpy()
+    if z.ndim == 1:
+        z = z.reshape(-1, 1)
+    if X is not None:
+        z = np.column_stack([z, X.to_numpy()])
+    return z
 
 
-class BaseReducer(_BaseWindowForecaster):
+def _check_fh(fh):
+    """Helper function to check fh for sliding-window transform"""
+    assert fh.is_relative
+    assert fh.is_all_out_of_sample()
+    return fh.to_indexer().to_numpy()
+
+
+def _sliding_window_transform(
+    y, window_length, fh, X=None, scitype="tabular-regressor"
+):
+    """Transform time series data `y` and `X` using sliding window.
+
+    Parameters
+    ----------
+    y : pd.Series
+        Endogenous time series
+    window_length : int
+        Window length for transformed feature variables
+    fh : ForecastingHorizon
+        Forecasting horizon for transformed target variable
+    X : pd.DataFrame, optional (default=None)
+        Exogenous series.
+    scitype : str {"tabular-regressor", "time-series-regressor"}, optional
+        Scitype of estimator to use with transformed data.
+        - If "tabular-regressor", returns X as tabular 2d array
+        - If "time-series-regressor", returns X as panel 3d array
+
+    See section 2.1 of this hackmd file for intended design:
```

Comment:
Remove link to hackmd - if we want to include this discussion, we should add it to the docstring or user guide.

Reply:
FYI @Lovkush-A